var/spack/repos/builtin/packages/suite-sparse/package.py

```diff
@@ -35,6 +35,8 @@ class SuiteSparse(Package):
     variant('cuda', default=False, description='Build with CUDA')
     variant('openmp', default=False, description='Build with OpenMP')
 
+    depends_on('mpfr', type=('build', 'link'))
+    depends_on('gmp', type=('build', 'link'))
```

Comment:
Shouldn't this dependency be restricted to the newer versions of suitesparse - i.e:
```
depends_on('mpfr', type=('build', 'link'), when='@5.8.0:')
depends_on('gmp', type=('build', 'link'), when='@5.8.0:')
```

Reply:
I don't know. Should it? If you say yes, I'll fix this and submit a PR to add this `when` clause.

Thank you!

Reply:
Quick test: build both 5.8 and 5.7 and run `ldd -r` (`otool -L` on macOS) on the executables and see if they link to `mpfr` and `gmp` or not.

Reply:
And just installed suite-sparse@5.7.2 and it does not link to mpfr or gmp.